cityofaustin/atd-vz-data
# Update CR3 download script

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/12849

Most of this is formatting with a small change to how the cookie is obtained in the script. No more prompt, and this script is now split out from the other CR3 tasks bundled in `atd-etl/app`. The script is now built and pushed to its own Docker image that is used by Airflow v2. The companion Airflow PR is https://github.com/cityofaustin/atd-airflow/pull/133.

## Testing
**URL to test:** 
None

**Steps to test:**
Trigger the DAG in the other Airflow PR or run locally by following the steps in the readme added in this diff. If you decide to run locally, the secrets for `.env` come out of the 1PW entries called **CRIS CR3 Download** and ~**Vision Zero CRIS Import**~ **Vision Zero graphql-engine Endpoints**. The cookie will also need to be added to the env file from CRIS.

You should see the script find no new pdfs to download and store in S3 unless you run it in the morning before the designated CRIS downloader runs the current process. 

If you want to go the extra mile, you can: 
- find an older crash record without serious injuries or fatalities (we don't want to disrupt VZ team efforts)
- carefully, [SSH tunnel and connect to the production database](https://app.gitbook.com/o/-LzDQOVGhTudbKRDGpUA/s/-MIQvl_rKnZ_-wHRdp4J/dev-guides/how-tos/how-to-connect-to-the-rds-instance#2.-open-the-tunnel) and update the one crash record's `cr3_stored_flag` column from `Y` to `N` (i tested this with crash id `16302089`)
- run the script and it will log one pdf download from CRIS and upload to S3

---
#### Ship list
